src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -386,6 +397,22 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    //This method validated the pattern of CMEK Key resource ID.
+    private void validateCmekKey(FailureCollector failureCollector) {
+      String[] part = cmekKey.split("/");
+      if ((part.length != 8) || !part[0].equals("projects") || !part[2].equals("locations")
```

Comment:
nit: when doing comparisons, it's better to put the constant first to avoid possible null pointer exceptions ("projects".equals(part[0])